src/js/core/game_state.ts

```diff
@@ -262,12 +251,12 @@ export class GameState {
 
     /**
      * Internal callback when entering the state. Do not override!
-     * @param {any} payload Arbitrary data passed from the state which we are transferring from
-     * @param {boolean} callCallback Whether to call the onEnter callback
+     * @param payload Arbitrary data passed from the state which we are transferring from
+     * @param callCallback Whether to call the onEnter callback
      */
-    internalEnterCallback(payload, callCallback = true) {
+    internalEnterCallback(payload: any, callCallback = true) {
```

Comment:
why's this `payload` typed with `any` while the one above was re-typed to `{}`?

Reply:
Well, technically it should be `object` in all instances, forcing all usages to explicitly check for property existence (this is correct, as states can be entered with any or no payload). If a payload is required, there should be an explicit assert/throw. I can look into changing all payload usages to `object`.

Reply:
ah I guess each usage having its own typing makes it weird, up to you then

Reply:
I have decided not to fiddle with this just yet. I have a few ideas in mind, but they're not perfect and I don't have good enough understanding of this shapez code area. If you have any improvements, I'd be thankful for PRs and concrete suggestions!